File: backend/app/schemas/evaluation.py
from pydantic import BaseModel
from typing import List, Dict, Optional

class DomainScoresSchema(BaseModel):
    classification: float
    manufacturer: float
    brand: float
    lov_compliance: float
    uom_compliance: float
    character_compliance: float
    completeness: float

class DiscrepancyItemSchema(BaseModel):
    mfg_part_num: str
    field_name: str
    category: str
    expected_value: str
    predicted_value: str
    status: str
    auto_fixable: bool

class MetricCategoryBreakdownSchema(BaseModel):
    category_name: str
    count: int
    percentage: float
    description: str

class EvaluationResultSchema(BaseModel):
    run_id: str
    eval_id: Optional[str] = None
    dataset_name: str
    products_evaluated: int
    total_evaluated: Optional[int] = 0
    overall_accuracy: float
    precision_score: Optional[float] = 0.95
    recall_score: Optional[float] = 0.92
    f1_score: float
    rouge_l_score: float
    lov_compliance_rate: float
    uom_compliance_rate: float
    auto_fix_success_rate: float
    autofix_success_rate: Optional[float] = 90.0
    character_compliance_rate: Optional[float] = 92.0
    classification_accuracy: Optional[float] = 95.0
    completeness_rate: Optional[float] = 90.0
    domain_scores: DomainScoresSchema
    category_counts: Dict[str, int]
    discrepancies: List[DiscrepancyItemSchema]
    created_at: str

File: frontend/types/evaluation.ts
export interface DiscrepancyItem {
  product_id: string;
  mpn: string;
  field_name: string;
  category: 'EXACT' | 'NORM' | 'FMT_MISMATCH' | 'MFR_MISMATCH' | 'BRAND_MISMATCH' | 'CLASS_MISMATCH' | 'UOM_MISMATCH' | 'MISSING' | 'EXTRA' | 'WRONG';
  expected_value: string;
  predicted_value: string;
  fix_available: boolean;
  status: string;
}

export interface DomainScores {
  identifiers: number;
  brand_normalization: number;
  taxonomy_classification: number;
  attribute_extraction: number;
  description_generation: number;
  digital_assets: number;
}

export interface MetricCategoryBreakdown {
  EXACT: number;
  NORM: number;
  FMT_MISMATCH: number;
  MFR_MISMATCH: number;
  BRAND_MISMATCH: number;
  CLASS_MISMATCH: number;
  UOM_MISMATCH: number;
  MISSING: number;
  EXTRA: number;
  WRONG: number;
}

export interface EvaluationResult {
  eval_id: string;
  timestamp: string;
  products_evaluated: number;
  total_evaluated: number;
  overall_accuracy: number;
  precision_score: number;
  recall_score: number;
  f1_score: number;
  rouge_l_score: number;
  lov_compliance_rate: number;
  uom_compliance_rate: number;
  character_compliance_rate: number;
  classification_accuracy: number;
  completeness_rate: number;
  autofix_success_rate: number;
  auto_fix_success_rate?: number;
  domain_scores: DomainScores;
  category_counts: MetricCategoryBreakdown;
  discrepancies: DiscrepancyItem[];
  is_demo_data?: boolean;
}
